src/lib/components/custom/dropdown.svelte

```diff
@@ -0,0 +1,100 @@
+<script lang="ts">
+	type Props = {
+		disabled?: boolean
+		up?: boolean
+		left?: boolean
+	}
+
+	let { disabled = false, up = false, left = false }: Props = $props()
+
+	let showDropdown = $state(false)
+	let dropdownElement: HTMLElement
+	let dropdownId: string
+
+	function close(ev: MouseEvent) {
+		const target = ev.target as unknown as Node
+		if (dropdownElement.contains(target)) {
+			// Clicked on the dropdown button or inside the dropdown
+		} else {
+			// Clicked outside the dropdown
+			showDropdown = false
+		}
+	}
+
+	function onKeyUp(ev: KeyboardEvent) {
+		if (ev.key === 'Escape') {
+			showDropdown = false
+		}
+	}
+
+	$effect(() => {
+		window.addEventListener('click', close)
+		window.addEventListener('keyup', onKeyUp)
+
+		return () => {
+			window.removeEventListener('click', close)
+			window.removeEventListener('keyup', onKeyUp)
+		}
+	})
+
+	function onClick() {
+		if (!disabled) showDropdown = !showDropdown
+	}
+</script>
+
+<div
+	bind:this={dropdownElement}
+	class="dropdown"
+	role="combobox"
+	aria-haspopup="listbox"
+	aria-expanded={showDropdown}
+	aria-controls={dropdownId}
+>
+	<div on:click={onClick} on:keypress={onClick} role="button" tabindex={0}>
+		<slot name="button" disabled active={showDropdown} />
+	</div>
+
+	<div class={`root`} aria-hidden={!showDropdown}>
+		<div
+			class={`${showDropdown ? '' : 'hidden'} ${up ? 'up' : ''} ${left ? 'left' : ''}`}
```

Comment:
This can be rewritten to a cleaner syntax
```suggestion
			class:hidden={showDropdown}
			class:up
			class:left
```